framework/src/fwk_status.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2019, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Description:
+ *     Framework status code helper.
+ */
+
+#include <fwk_status.h>
+#include <fwk_macros.h>
+
+const char *fwk_status_str(int status)
+{
+    switch (status) {
+    case FWK_PENDING:
+        return "PENDING";
+    case FWK_SUCCESS:
+        return "SUCCESS";
+    case FWK_E_PARAM:
+        return "E_PARAM";
+    case FWK_E_ALIGN:
+        return "E_ALIGN";
+    case FWK_E_SIZE:
+        return "E_SIZE";
+    case FWK_E_HANDLER:
+        return "E_HANDLER";
+    case FWK_E_ACCESS:
+        return "E_ACCESS";
+    case FWK_E_RANGE:
+        return "E_RANGE";
+    case FWK_E_TIMEOUT:
+        return "E_TIMEOUT";
+    case FWK_E_NOMEM:
+        return "E_NOMEM";
+    case FWK_E_PWRSTATE:
+        return "E_PWRSTATE";
+    case FWK_E_SUPPORT:
+        return "E_SUPPORT";
+    case FWK_E_DEVICE:
+        return "E_DEVICE";
+    case FWK_E_BUSY:
+        return "E_BUSY";
+    case FWK_E_OS:
+        return "E_OS";
+    case FWK_E_DATA:
+        return "E_DATA";
+    case FWK_E_STATE:
+        return "E_STATE";
+    case FWK_E_INIT:
+        return "E_INIT";
+    case FWK_E_OVERWRITTEN:
+        return "E_OVERWRITTEN";
+    case FWK_E_PANIC:
+        return "E_PANIC";
+    default:
+        break;
+    }
+
+    if (status < 0)
+        return "unknown-negative";
+
+    return "unknown-positive";
```

Comment:
I think it would suffice to simply:

```
    default:
        return "unknown";
```

As I expect anywhere where it actually matters is going to print out the status code anyway. Opinions?

Reply:
you're right. this distinction is not really useful. i'll fix.